### PR TITLE
Updates deprecated functions, fixed examples

### PR DESCRIPTION
**Pull request by michel2323**
* Updated function setObjective -> JuMP.setobjective
* Updated macros @defVar -> @variable and @constraint
* Fixed simple_qd_test

I'm a total newbie and will probably be going through other examples.

### CODE REVIEW

**Comment by cnpetra** [comment]
hi Michel, welcome on board!

**Comment by michel2323** [comment]
Hi! Expect some silly questions...

File: examples/easy_test.jl [hunk 1]
```diff
@@ -3,15 +3,15 @@ using StructJuMP
...
 
-@defVar(m, 0 <= x <= 1)
-@defVar(m, 0 <= y <= 1)
+@variable(m, 0 <= x <= 1)
+@variable(m, 0 <= y <= 1)
 
-@addConstraint(m, x + y == 1)
-setObjective(m, :Min, x*x + y)
+@constraint(m, x + y == 1)
+JuMP.setobjective(m, :Min, x*x + y)
```
Comment (joehuchette) [inline examples/easy_test.jl]: This should be ``@setobjective`` instead.

File: examples/easy_test.jl [hunk 2]
```diff
@@ -3,15 +3,15 @@ using StructJuMP
...
 
 for i in 1:numScen
     bl = StructuredModel(parent=m)
-    @defVar(bl, w >= 0)
-    @addConstraint(bl, w - x - y <= 1)
-    setObjective(bl, :Min, w*w + w)
+    @variable(bl, w >= 0)
+    @constraint(bl, w - x - y <= 1)
+    JuMP.setobjective(bl, :Min, w*w + w)
```
Comment (joehuchette) [inline examples/easy_test.jl]: Same as above

File: examples/explicit_stochastic.jl [hunk 3]
```diff
@@ -21,7 +21,7 @@ m = StructuredModel()
 
 s0 = StructuredModel(parent=m)
 @defStochasticVar(s0, 0 <= y[TESTTIME,GEN] <= 1)
-@setObjective(s0, Min, FIXEDGENCOST*sum{y[t,j], t=TESTTIME,j=GEN})
+@JuMP.setobjective(s0, Min, FIXEDGENCOST*sum{y[t,j], t=TESTTIME,j=GEN})
```
Comment (joehuchette) [inline examples/explicit_stochastic.jl]: ditto

File: examples/explicit_stochastic.jl [hunk 4]
```diff
@@ -31,38 +31,38 @@ for it in 1:NUMSTAGES
...
     for t in TESTTIME
         for j in GEN
-            @addConstraint(st, -np_cap[j] <= Pgen[t,j] - np_cap[j]*y[t,j] <= 0) # should wrap the call to y in an ancestor-type thing
+            @constraint(st, -np_cap[j] <= Pgen[t,j] - np_cap[j]*y[t,j] <= 0) # should wrap the call to y in an ancestor-type thing
         end
     end
 
     # need Exp here for earlier sml versions
-    @setObjective(st, Min, sum{COSTSCALE*gen_cost[i]*Pgen[t,i], t=TESTTIME, i=GEN})
+    @JuMP.setobjective(st, Min, sum{COSTSCALE*gen_cost[i]*Pgen[t,i], t=TESTTIME, i=GEN})
```
Comment (joehuchette) [inline examples/explicit_stochastic.jl]: here as well

File: examples/explicit_stochastic_sml.jl [hunk 5]
```diff
@@ -25,17 +25,17 @@ for n1 in NODES
...
-                @addConstraint(sub, xh2[j] == (1+Ret2[n2,j])*xh1[j] + xb2[j] - xs2[j])
+                @constraint(sub, xh2[j] == (1+Ret2[n2,j])*xh1[j] + xb2[j] - xs2[j])
             end
-            @addConstraint(sub, (1-Ct)*sum{val[j]*xs2[j], j=ASSETS} == Liability2 + (1+Ct)*sum{val[j]*xb2[j]})
+            @constraint(sub, (1-Ct)*sum{val[j]*xs2[j], j=ASSETS} == Liability2 + (1+Ct)*sum{val[j]*xb2[j]})
             @defStochasticVar(sub, wealth)
-            @addConstraint(sub, wealth == Prob[n1]*Prob[n2]*sum{val[j]*xh2[j], j=ASSETS})
-            @setObjective(sub, mu - Rho*((wealth^2-mu^2))*Prob[n1]*Prob[n2])
+            @constraint(sub, wealth == Prob[n1]*Prob[n2]*sum{val[j]*xh2[j], j=ASSETS})
+            @JuMP.setobjective(sub, mu - Rho*((wealth^2-mu^2))*Prob[n1]*Prob[n2])
```
Comment (joehuchette) [inline examples/explicit_stochastic_sml.jl]: same here

File: examples/harder_test.jl [hunk 6]
```diff
@@ -3,12 +3,12 @@ using StructJuMP, MPI
...
-@defVar(m, y <= 2)
+@variable(m, x >= 1)
+@variable(m, y <= 2)
 
-@addConstraint(m,  x + y == 1)
-@addConstraint(m, -x + y <= 1)
-setObjective(m, :Min, x*x + 0.5x*y + 0.25y*y - y)
+@constraint(m,  x + y == 1)
+@constraint(m, -x + y <= 1)
+JuMP.setobjective(m, :Min, x*x + 0.5x*y + 0.25y*y - y)
```
Comment (joehuchette) [inline examples/harder_test.jl]: also

File: examples/harder_test.jl [hunk 7]
```diff
@@ -17,8 +17,8 @@ ac   = [1,0.75]
...
 @second_stage m scen begin
     bl = StructuredModel(parent=m)
-    @defVar(bl, 0 <= w <= 1)
-    @addConstraint(bl, coef[scen]w - x - y <= rhs[scen])
-    @addConstraint(bl, coef[scen]w + x     == rhs[scen])
-    setObjective(bl, :Min, qc[scen]*w*w + ac[scen]*w)
+    @variable(bl, 0 <= w <= 1)
+    @constraint(bl, coef[scen]w - x - y <= rhs[scen])
+    @constraint(bl, coef[scen]w + x     == rhs[scen])
+    JuMP.setobjective(bl, :Min, qc[scen]*w*w + ac[scen]*w)
```
Comment (joehuchette) [inline examples/harder_test.jl]: also

File: examples/large_simple_test.jl [hunk 8]
```diff
@@ -5,11 +5,11 @@ numScen = 1
 
 m = StructuredModel(numScen)
 
-@defVar(m, x[1:N] >= 0)
-@addConstraint(m, sum{x[i], i=1:N} >= -1)
-@setObjective(m, Min, sum{x[i], i=1:N})
+@variable(m, x[1:N] >= 0)
+@constraint(m, sum{x[i], i=1:N} >= -1)
+@JuMP.setobjective(m, Min, sum{x[i], i=1:N})
```
Comment (joehuchette) [inline examples/large_simple_test.jl]: same

File: examples/large_simple_test.jl [hunk 9]
```diff
@@ -5,11 +5,11 @@ numScen = 1
...
 
 bl = StructuredModel(parent=m)
-@defVar(bl, y[1:N] >= 0)
-@addConstraint(bl, constr[r=2:(N-1)], x[r] + x[r-1] + y[r] + y[r+1] >= 0)
-@setObjective(bl, Min, sum{y[i], i=1:N})
+@variable(bl, y[1:N] >= 0)
+@constraint(bl, constr[r=2:(N-1)], x[r] + x[r-1] + y[r] + y[r+1] >= 0)
+@JuMP.setobjective(bl, Min, sum{y[i], i=1:N})
```
Comment (joehuchette) [inline examples/large_simple_test.jl]: again

File: examples/stochastic.jl [hunk 10]
```diff
@@ -8,20 +8,20 @@ m = StructuredModel()
...
 
 sub = StructuredModel(parent=m)
-@defVar(sub, wealth)
-@addConstraint(sub, wealth == sum{val[j]*xh[j], j=ASSETS})
-@addConstraint(sub, @Exp(wealth) == mu)
-setObjective(sub, mu - Rho*((wealth*wealth)- mu*mu))
+@variable(sub, wealth)
+@constraint(sub, wealth == sum{val[j]*xh[j], j=ASSETS})
+@constraint(sub, @Exp(wealth) == mu)
+JuMP.setobjective(sub, mu - Rho*((wealth*wealth)- mu*mu))
```
Comment (joehuchette) [inline examples/stochastic.jl]: here as well

File: examples/transportation.jl [hunk 11]
```diff
@@ -31,20 +31,20 @@ wastecost = 4
 @defStochasticVar(m, profit)
 
 @defStochasticVar(m, received[centers] >= 0)
-@setObjective(m, Max, -sum{transcost[i,j]*ship[i,j], i=factories, j=centers} + sum{prodcost*product[i], i=factories})
+@JuMP.setobjective(m, Max, -sum{transcost[i,j]*ship[i,j], i=factories, j=centers} + sum{prodcost*product[i], i=factories})
```
Comment (joehuchette) [inline examples/transportation.jl]: almost the last one...

File: examples/stochpdegas.jl [hunk 12]
```diff
@@ -25,78 +25,78 @@ dt = Tf / Nt
...
+@variable(m, 0 <= phi[SCEN])
+@variable(m, cvarcost)
 
 for k in SCEN
-    @addConstraint(m, cost[k] - nu <= phi[k])
+    @constraint(m, cost[k] - nu <= phi[k])
 end
 
-@setObjective(m, (1-lambda)*mcost + lambda*cvarcost)
+@JuMP.setobjective(m, (1-lambda)*mcost + lambda*cvarcost)
```
Comment (joehuchette) [inline examples/stochpdegas.jl]: again

File: examples/transportation.jl [hunk 13]
```diff
@@ -31,20 +31,20 @@ wastecost = 4
...
-    @addConstraint(m, product[i] == sum{ship[i,j], j=centers})
+    @constraint(m, product[i] == sum{ship[i,j], j=centers})
 end
 
 for (s, elem) in enumerate(scenarios)
     bl = StructuredModel(parent=m)
     @defStochasticVar(bl, 0 <= salesw[i=centers] <= demand[i,s])
     @defStochasticVar(bl, wastew[centers] >= 0)
-    @setObjective(bl, Max, sum{price*prob[s]*salesw[j], j=centers} - sum{wastecost*prob[s]*wastew[j], j=centers})
+    @JuMP.setobjective(bl, Max, sum{price*prob[s]*salesw[j], j=centers} - sum{wastecost*prob[s]*wastew[j], j=centers})
```
Comment (joehuchette) [inline examples/transportation.jl]: last one!

**Comment by ktarplee** [comment]
#38 cherry-picked this commit so this content is already on master

**Comment by joehuchette** [comment]
Thanks, @michel2323!

**Pull request closed without merging**